docs/source/serving/high-availability.rst

```diff
@@ -0,0 +1,207 @@
+      # not if you have an existing one (even if stopped).
+      controller:
+        # --- Controller Resources ---
+        # Optional: Specify resources like cloud, region, cpus, disk_size
+        resources:
+          cloud: kubernetes  # High Availability mode requires Kubernetes
+          # region: <your-k8s-region> # Optional
+          cpus: 2+           # Optional, example value
+          # disk_size: 100     # Optional, example value (affects PVC size in High Availability mode)
+
+        # --- Enable High Availability ---
+        high_availability: true
+
+.. note::
+    Enabling or disabling ``high_availability`` only affects **new** SkyServe controllers. If you have an existing controller (either running or stopped), changing this setting will not modify it. To apply the change, you must first terminate all services and then tear down the existing controller using ``sky down <controller_name>``. See `Important considerations`_ below.
+
+.. note::
+    While the controller requires Kubernetes for High Availability mode, service replicas can still use any cloud or infrastructure supported by SkyPilot.
+
+How it works
+------------
+When ``high_availability: true`` is set, SkyPilot modifies how the SkyServe controller is deployed on Kubernetes:
+
+The High Availability implementation relies on standard Kubernetes mechanisms to ensure controller resilience:
+
+* **Automatic Recovery:** The controller runs as a Kubernetes Deployment that automatically restarts pods after failures.
+* **Persistent State:** Critical controller state (database, configuration) is stored on persistent storage that persists on pod restarts.
+* **Seamless Continuation:** When a new pod starts after a failure, it automatically reconnects to existing resources and continues operations without manual intervention.
+
+The entire recovery process is handled transparently by SkyPilot and Kubernetes, requiring no action from users when failures occur.
+
+Configuration details
+---------------------
+Besides the main ``serve.controller.high_availability: true`` flag, you can customize High Availability behavior further:
+
+.. raw:: html
+
+   <ul>
+   <li><strong>Controller Resources (<code>serve.controller.resources</code>):</strong> As usual, you can specify <code>cloud</code> (must be Kubernetes), <code>region</code>, <code>cpus</code>, etc. The <code>disk_size</code> here directly determines the size of the PersistentVolumeClaim created for the High Availability controller.</li>
+   <li><strong>Kubernetes Storage Class (<code>kubernetes.high_availability.storage_class_name</code> - Optional):</strong> If your Kubernetes cluster has specific storage classes defined (e.g., for different performance tiers like SSD vs HDD, or specific features like backup), you can specify which one to use for the controller's PVC. This is configured under the <code>kubernetes</code> section in <code>config.yaml</code>:</li>
+   </ul>
```

Comment:
Hmm, why do we use raw HTML here?

Reply:
Since the code block nested in bold block won't be rendered. https://stackoverflow.com/questions/26366552/bold-code-in-rst

Reply:
Or let's just skip stressing this `kubernetes.high_availability.storage_class_name`, and mention it in the normal text?